test/seqan3-test.cmake

```diff
@@ -48,22 +48,25 @@ include (FindPackageMessage)
 set (SEQAN3_BENCHMARK_CLONE_DIR "${PROJECT_BINARY_DIR}/vendor/benchmark")
 set (SEQAN3_TEST_CLONE_DIR "${PROJECT_BINARY_DIR}/vendor/googletest")
 
-# required flags, includes, definitions and libraries for seqan3
-set (SEQAN3_STRICT_CXX_FLAGS "-pedantic" "-Werror" "-Wall" "-Wextra")
+# required flags, includes, definitions and libraries for seqan3 tests
+add_library (seqan3_test INTERFACE)
+target_link_libraries (seqan3_test INTERFACE "seqan::seqan3")
+target_link_libraries (seqan3_test INTERFACE "pthread")
+target_compile_options (seqan3_test INTERFACE "-pedantic" "-Werror" "-Wall" "-Wextra")
+target_include_directories (seqan3_test INTERFACE "${SEQAN3_CLONE_DIR}/test/include/")
```

Comment:
Did I understand this correctly:
we define a new "library interface" that depends on another interface "seqan::seqan3", but also has additional properties?

Reply:
Yep, it's transitive. https://cmake.org/cmake/help/v3.0/manual/cmake-buildsystem.7.html#transitive-usage-requirements